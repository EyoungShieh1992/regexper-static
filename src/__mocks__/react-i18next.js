const reactI18next = jest.requireActual('react-i18next');
const i18n = require('i18n');

module.exports = {
  ...reactI18next,
  Trans: require('__mocks__/component-mock').buildMock(reactI18next.Trans),
  useTranslation: () => ({ i18n, t: i18n.mockT })
};
