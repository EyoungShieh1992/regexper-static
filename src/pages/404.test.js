jest.mock('components/Metadata', () =>
  require('__mocks__/component-mock')('components/Metadata'));
jest.mock('components/Message', () =>
  require('__mocks__/component-mock')('components/Message'));

import React from 'react';
import { render } from 'react-testing-library';

import ErrorPage from 'pages/404';

describe('Error Page', () => {
  test('rendering', () => {
    const { asFragment } = render(
      <ErrorPage/>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
