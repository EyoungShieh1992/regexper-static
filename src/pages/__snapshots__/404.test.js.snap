// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page rendering 1`] = `
<DocumentFragment>
  <span
    data-component="Metadata"
    data-props="{
  \\"title\\": \\"TRANSLATE(Page Not Found)\\"
}"
  />
  <span
    data-component="Message"
    data-props="{
  \\"type\\": \\"error\\",
  \\"heading\\": \\"TRANSLATE(404 Page Not Found)\\"
}"
  >
    <p>
      <span
        data-component="Trans"
        data-props="{}"
      >
        The page you have requested could not be found.
      </span>
    </p>
  </span>
</DocumentFragment>
`;
